code/game/objects/items/weapons/material/material_armor.dm

```diff
@@ -249,6 +323,47 @@ Protectiveness | Armor %
 	else
 		..()
 
+//Make plating inserts for modular armour.
+/obj/item/weapon/material/armor_plating/insert/attackby(var/obj/O, mob/user)
+	if(istype(O, /obj/item/weapon/weldingtool))
+		var /obj/item/weapon/weldingtool/S = O
+		if(S.remove_fuel(0,user))
+			if(!src || !S.isOn()) return
+			to_chat(user, "<span class='notice'>You trim down the edges to size.</span>")
+			user.drop_from_inventory(src)
+			var/obj/item/clothing/accessory/material/makeshift/light/new_armor = new(null, src.material.name)
+			user.put_in_hands(new_armor)
+			qdel(src)
+
+	if(istype(O, /obj/item/weapon/material/armor_plating/insert))
+		var/obj/item/weapon/material/armor_plating/insert/second_plate = O
+		if(second_plate.material != src.material)
+			to_chat(user, "<span class='warning'>Both plates need to be the same type of material.</span>")
+			return
+		to_chat(user, "<span class='notice'>You bond the two plates together.</span>")
+		user.drop_from_inventory(src)
+		user.drop_from_inventory(second_plate)
+		var/obj/item/clothing/accessory/material/makeshift/heavy/new_armor = new(null, src.material.name)
+		user.put_in_hands(new_armor)
+		qdel(second_plate)
+		qdel(src)
+
+	if(istype(O, /obj/item/weapon/tool/wirecutters))
+		to_chat(user, "<span class='notice'>You split the plate down the middle, and joint it at the elbow.</span>")
+		user.drop_from_inventory(src)
+		var/obj/item/clothing/accessory/material/makeshift/armguards/new_armor = new(null, src.material.name)
+		user.put_in_hands(new_armor)
+		qdel(src)
+
+	if(istype(O, /obj/item/stack/material))
+		var/obj/item/stack/material/S = O
+		if(S.material == get_material_by_name("leather"))
+			if(S.use(2))
+				to_chat(user, "<span class='notice'>You curve the plate inwards, and add a strap for adjustment.</span>")
+				user.drop_from_inventory(src)
+				var/obj/item/clothing/accessory/material/makeshift/legguards/new_armor = new(null, src.material.name)
+				user.put_in_hands(new_armor)
+				qdel(src)
```

Comment:
You need to call `. = ..()` at the end of this proc to ensure you get all the behavior for other items like pens/labellers/grippers/etc.